RHDEVDOCS-4485 migration guide - odo v2 to v3 (#51479)
openshift/openshift-docs: modules/odo-debugging-changes.adoc

// Module included in the following assemblies:
//
// * cli_reference/developer_cli_odo/odo-migration-guide.adoc

:_content-type: CONCEPT
[id="odo-debugging-changes_{context}"]

= Changes to component debugging

> With `odo` v2, you use the `odo push --debug` command to run a component in debug mode. To set up port forwarding to the component's debug port, you have to run `odo debug port-forward`.

Reviewer: Maybe I'm overthinking this, but would this content make sense as a small procedure? Something like:

= Changing component debugging 
For both `odo` v2 and v3, you can debug your components. 

== Debugging components in `odo` v2

1. Run a component in debug mode by running, `odo push --debug`.
2. Set up port forwarding to the component's debug port by running, ``odo debug port-forward`.

== Debugging components in `odo` v3

1. Specify the debug port in the `devfile.yaml` as an endpoint.
2.  Run `odo dev --debug`.
3. Start the component in debug mode. 

The following example shows a `container` component in the Devfile, where port `3000` is the application port and `5858` is the debug port: (provide your sample devfile yaml that you have)

Author: That would be good, but would require that I have the YAML/config for V2 which is different to what is available for V3. SO I am going to leave it as is, just in case I screw it up with the wrong set of steps

> With `odo` v2, you use the `odo push --debug` command to run a component in debug mode. To set up port forwarding to the component's debug port, you have to run `odo debug port-forward`.

Reviewer 2: ```suggestion
In `odo` v2, you had to use the `odo push --debug` command to run a component in debug mode. To set up port forwarding to the component's debug port, you had to run `odo debug port-forward`.
```

> In `odo` v3, you specify the debug port in the `devfile.yaml` as an endpoint, and run the `odo dev --debug` command to start the component in debug mode. The following example shows a `container` component in the Devfile, where port `3000` is the application port and `5858` is the debug port:

Reviewer 2: ```suggestion
With `odo` v3, you can specify the debug port in the `devfile.yaml` as an endpoint, and run the `odo dev --debug` command to start the component in debug mode, the port forwarding happens by default in this case. The following example shows a `container` component in the Devfile, where port `3000` is the application port and `5858` is the debug port:
```
Or something like that.
